### PR TITLE
Eric s geo coder

### PR DESCRIPTION
**Pull request by ESmith92**
update url builder
    -rename google_builder() to reverse_google_builder()
    -add forward_google_builder()
    -add address variable
    -add set_address()
    -update function name in test and google-revrse-geocoder class

update google-reverse-geocoder
    -rename get_url() to get_reverse_url()
    -add get_forward_url()
    -move service call and json decode to function get_data()
    -update get_locations() to use get_data()
    -add get_coordinates() to get coords from a "address"
    -update spec test to use get_reverse_url()
    -update implementation to force new functions()

create parser for the forward geo coding
    -add class to loop through the json from google forward geo coding
        -loops through results and selects the coordinates
        -checks if coordinate is already in the array
        -returns an array of GeoCoordinates
    -update google-reverse-geocoder to use the forward parser for the forward geo coding

create test for the forward geo coding
    -create unit testing to check that forward geo parser is initalizable
    -create spec tests for the forward geo coding
        -update googleReverseGeocoderSpec to have test for the forward geo coding
            -add test to check url builder returned url
            -add test to run through the process of forward geo coding
        -create spec tests for the forward geo coding parser
    -create integration testing with google's forward geo coding service

### CODE REVIEW

File: src/providers/google-reverse-geocoder.php [hunk 1]
```diff
@@ -29,7 +29,8 @@ class GoogleReverseGeocoder implements ReverseGeocoder {
   function __construct( Support\GoogleURLBuilder $url_builder = null,
     Commands\HttpGet $curl_caller = null,
     Support\JsonDecoder $geo_json_decoder = null,
-    Support\GoogleReverseGeocoderParser $geo_parser = null ) {
+    Support\GoogleReverseGeocoderParser $geo_parser = null,
+    Support\GoogleForwardGeocoderParser $geo_parser2 = null ) {
```
Comment (rraub) [inline src/providers/google-reverse-geocoder.php]: geo_parser2.. really.. shorter variable names are not faster, they are just more confusing. 😕

**Comment by rraub** [comment]
Sorry it took me so long to look at this!

**Comment by ESmith92** [comment]
It broke while composer was updating and I don't know how to rerun the test without just making a new pull request

**Comment by rraub** [comment]
I restarted the tests for you, thanks for making the corrections!

**Pull request merged**